src/cascadia/ShellExtension/OpenTerminalHere.cpp

```diff
@@ -204,14 +204,14 @@ HRESULT OpenTerminalHere::GetBestLocationFromSelectionOrSite(IShellItemArray* ps
     return S_OK;
 }
 
-// This method checks if any of the ctrl keys are pressed during activation of the shell extension
-bool OpenTerminalHere::IsControlPressed()
+// Check is both ctrl and shift keys are pressed during activation of the shell extension
+bool OpenTerminalHere::IsControlAndShiftPressed()
 {
-    const auto ControlPressed = 1U;
+    short control = 0;
+    short shift = 0;
+    control = GetAsyncKeyState(VK_CONTROL);

```

Comment:
qq: Why the switch from sync to Async?

Reply:
GetKeyState is really intended to be used by an application with a message queue which I dont think is the case here with a shell extension. After some searching and alot of testing different ideas this is where I landed that was reliable.

Reply:
I would guess that's the UI thread of file explorer?

Reply:
So, this is where it gets weird. When we are in the Windows Terminal package our shell extension gets hosted in a separate process called `dllhost.exe`!